spec/core/v2/ics-004-packet-semantics/README.md

```diff
@@ -206,513 +196,111 @@ function channelCapabilityPath(portIdentifier: Identifier, channelIdentifier: Id
-
-When the opening handshake is complete, the module which initiates the handshake will own the end of the created channel on the host ledger, and the counterparty module which
-it specifies will own the other end of the created channel on the counterparty chain. Once a channel is created, ownership cannot be changed (although higher-level abstractions
-could be implemented to provide this).
-
-Chains MUST implement a function `generateIdentifier` which chooses an identifier, e.g. by incrementing a counter:
-
-```typescript
-type generateIdentifier = () -> Identifier
-```
-
-```typescript
-function chanOpenInit(
-  order: ChannelOrder,
-  connectionHops: [Identifier],
-  portIdentifier: Identifier,
-  counterpartyPortIdentifier: Identifier): (channelIdentifier: Identifier, channelCapability: CapabilityKey) {
-    channelIdentifier = generateIdentifier()
-    abortTransactionUnless(validateChannelIdentifier(portIdentifier, channelIdentifier))
-
-    abortTransactionUnless(provableStore.get(channelPath(portIdentifier, channelIdentifier)) === null)
-    connection = provableStore.get(connectionPath(connectionHops[0]))
-
-    // optimistic channel handshakes are allowed
-    abortTransactionUnless(connection !== null)
-    abortTransactionUnless(authenticateCapability(portPath(portIdentifier), portCapability))
-
-    channelCapability = newCapability(channelCapabilityPath(portIdentifier, channelIdentifier))
-    provableStore.set(nextSequenceSendPath(portIdentifier, channelIdentifier), 1)
-    provableStore.set(nextSequenceRecvPath(portIdentifier, channelIdentifier), 1)
-    provableStore.set(nextSequenceAckPath(portIdentifier, channelIdentifier), 1)
-
-    return channelIdentifier, channelCapability
-}
-```
-
-The `chanOpenTry` function is called by a module to accept the first step of a channel opening handshake initiated by a module on another chain.
+Thus, IBC version 2 introduces a new message `RegisterCounterparty` that will associate the counterparty client of our chain with our client of the counterparty. Thus, if the `RegisterCounterparty` message is submitted to both sides correctly. Then both sides have mirrored <client,client> pairs that can be treated as channel identifiers. Assuming they are correct, the client on each side is unique and provides an authenticated stream of packet data between the two chains. If the `RegisterCounterparty` message submits the wrong clientID, this can lead to invalid behaviour; but this is equivalent to a relayer submitting an invalid client in place of a correct client for the desired chain. In the simplest case, we can rely on out-of-band social consensus to only send on valid <client, client> pairs that represent a connection between the desired chains of the user; just as we currently rely on out-of-band social consensus that a given clientID and channel built on top of it is the valid, canonical identifier of our desired chain.
```

Comment:
Good explanation!!! Let's update for having channel id and client id separate as well